dataflows/processors/join.py

```diff
@@ -324,16 +350,16 @@ def func(package: PackageWrapper):
     return func
 
 
-def join(source_name, source_key, target_name, target_key, fields={}, full=True, source_delete=True):
-    return join_aux(source_name, source_key, source_delete, target_name, target_key, fields, full)
+def join(source_name, source_key, target_name, target_key, fields={}, full=True, mode=None, source_delete=True):
```

Comment:
full should default to None and mode to 'inner'
if full is not None - show deprecation warning and update mode accordingly

Reply:
same comment applies to below functions as well